include/extern.h

```diff
@@ -534,7 +534,7 @@ E void NDECL(update_mlstmv);
 E void NDECL(losedogs);
 E void FDECL(mon_arrive, (struct monst *, BOOLEAN_P));
 E void FDECL(mon_catchup_elapsed_time, (struct monst *, long));
-E void FDECL(keepdogs, (BOOLEAN_P));
+E void FDECL(keepdogs, (BOOLEAN_P, BOOLEAN_P));
```

Comment:
I like the general idea but not in this form.

As stated in the commit message, there are situations when you
shouldn't be able to notice that monsters followed you.

I also seriously dislike just adding boolean flags to functions (note that the current code sometimes also does that so this is more of a general remark than just criticism of this commit).

One boolean is often okay, but two or more make it unreadable. Either
break it out into different functions or change the booleans to enum
options.

Reply:
When the booleans are totally independent, the only real options for reducing the number of arguments are to make a clunky enum that enumerates the different cases, or use bitflags, the added complication of which may be worth it for 3+ independent booleans but probably not 2.